tests/src/tools/ReadyToRun.TestHarness/Program.cs

```diff
@@ -60,7 +60,7 @@ private static ArgumentSyntax ParseCommandLine(string[] args)
                 syntax.HandleErrors = true;
                 syntax.HandleResponseFiles = true;
 
-                syntax.DefineOption("h|help", ref _help, "Help message for R2RDump");
+                syntax.DefineOption("h|help", ref _help, "Help message for TestHarness");
```

Comment:
Non-blocking - I was actually wondering whether there might be a way to unify your new SuperILC tooling with the TestHarness app. Leaving aside the fact that it's usually somewhat easier for maintenance to have a single app, it might nicely facilitate some of the scenarios I mentioned in the other comment - e.g. compiling a test with both CPAOT and Crossgen, running both versions and producing some diff of the ETW JIT info.

Reply:
We could even compile R2RDump (or some subset thereof) into the app and make it directly able to produce the static analysis diffs without having to run R2RDump as an external app.

Reply:
I personally prefer small, simple tools that don't try to do everything. Having a one-liner command to compile and run interesting scenarios is interesting though. I'll have a think about it.

Reply:
I'm not advocating artificial unification just to compile ten different tools into the same exe; of course this only makes sense if there is sufficient functional overlap among the tools. In your initial implementation there is not much overlap, my point was just that the two apps may be "close enough" so that it might make sense to create this overlap by having a tool that can compile and / or execute a test or app. Compiling in R2RDump is perhaps too much especially as its source code resides in a different repo, perhaps it would be reasonable to rather specify it as a path to the external tool. It would be also nice to think how all of this can work in a lab.

Reply:
Yeah - if we can make these things exposable as libraries that can fit into some pipeline of Compile -> Execute -> Dump Analysis I think we can keep the simple tools and get the nice emergent behavior of combining them.

Reply:
That is an interesting expansion on my original thinking - having some simple building blocks, either managed exe's or dll's acting as msbuild tasks, would be an interesting generalization that would let us create more advanced test scenarios very easily - e.g. compiling an app using CPAOT multiple times with different build flags and comparing such results, compiling a full 2 x 2 matrix of (Debug / Release) x (CPAOT / Crossgen).

For exe's, that's basically the current state - you have already implemented the "executor" (TestHarness) and now you're implementing a "compiler" (or rather something like a "selector" combined with a "compiler"). For dll's as msbuild tasks, the "selector" may be doable in msbuild so perhaps we need just the "compiler" and "executor". [I'm not sure about identifying managed assemblies.]

One way or another, I would love us to land in a place that would let us employ parallelization of CPAOT compilations of the individual assemblies. I believe it's a safe bet that compilation time of the ASP.NET or CoreCLR framework assemblies would get reduced to one quarter on my 4-processor devbox when parallelized.